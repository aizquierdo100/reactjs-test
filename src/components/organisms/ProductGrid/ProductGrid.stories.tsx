import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import ProductGrid from './ProductGrid';
import { EffectStatus } from "../../../models";
import { BrowserRouter } from 'react-router-dom';

export default {
    title: 'Components/Organisms/ProductGrid',
    component: ProductGrid,
    argTypes: {
        backgroundColor: { control: 'color' },
    },
} as ComponentMeta<typeof ProductGrid>;

const Template: ComponentStory<typeof ProductGrid> = (args) => <BrowserRouter><ProductGrid {...args} /></BrowserRouter>

export const ProductGridSuccess = Template.bind({});
ProductGridSuccess.args = {
    products: [{
        id: 4,
        sku: "C47R",
        name: "Stripe dress",
        description: 'Description',
        image: "model5.jpg",
        price: 300000,
    }],
    status: EffectStatus.SUCCESS,
    addCartHandler: () => {
        alert('add to cart')
    }
};

export const ProductGridError = Template.bind({});
ProductGridError.args = {
    products: [{
        id: 4,
        sku: "C47R",
        name: "Stripe dress",
        description: 'Description',
        image: "model5.jpg",
        price: 300000,
    }],
    status: EffectStatus.ERROR,
    addCartHandler: () => {
        alert('add to cart')
    }
};

export const ProductGridLoading = Template.bind({});
ProductGridLoading.args = {
    products: [{
        id: 4,
        sku: "C47R",
        name: "Stripe dress",
        description: 'Description',
        image: "model5.jpg",
        price: 300000,
    }],
    status: EffectStatus.LOADING,
    addCartHandler: () => {
        alert('add to cart')
    }
};


