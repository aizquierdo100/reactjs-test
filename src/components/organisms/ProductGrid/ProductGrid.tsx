import React, {FC} from "react";
import {useNavigate} from "react-router-dom";

import ProductCard from "../../molecules/ProductCard/ProductCard";
import '../../../assets/styles/grid.scss';
import '../../../assets/styles/animations.scss';
import {EffectStatus, IProduct} from "../../../models";
import {shuffle} from "../../../utils";

interface IProps {
    products: IProduct[],
    isLoading: EffectStatus
    addCartHandler: (id: number) => void;
}

const ProductGrid: FC<IProps> = ({products, isLoading, addCartHandler}) => {

    const navigate = useNavigate();

    const moreInfoHandler = (id: number): void => {
        navigate(`/details/${id}`);
    }

    const productsContent = isLoading ? products ? shuffle(products).slice(2).map((x) => {
        const product = x as IProduct;
        return (
            <React.Fragment key={product.id}>
                <div className='g--3 g-m--4 g-s--12 fade-in-from-top'>
                    <ProductCard
                        alt={product.name}
                        moreInfoHandler={() => moreInfoHandler(product.id)}
                        addCartHandler={() => addCartHandler(product.id)}
                        image={require(`../../../assets/images/products/${product.image}`)}
                        price={product.price}
                        title={product.name}/>
                </div>
            </React.Fragment>
        )
    }) : null : 'Loading';

    return (
        <div className='container container--around container--wrap--m'>
            {productsContent}
        </div>
    )
}

export default ProductGrid;