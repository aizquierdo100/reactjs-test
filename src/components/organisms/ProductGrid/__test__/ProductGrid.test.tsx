/**
 *
 * CategoryCard Test
 * @format
 *
 */

import { composeStories } from "@storybook/testing-react";
import * as Stories from "../ProductGrid.stories";
import { render } from "@testing-library/react";

const { ProductGridError, ProductGridLoading, ProductGridSuccess } = composeStories(Stories);

describe("ProductGrid", () => {
    test("renders ProductGridError correctly", () => {
        const view = render(<ProductGridError />);
        expect(view).toMatchSnapshot();
    });

    test("renders ProductGridLoading correctly", () => {
        const view = render(<ProductGridLoading />);
        expect(view).toMatchSnapshot();
    });

    test("renders ProductGridSuccess correctly", () => {
        const view = render(<ProductGridSuccess />);
        expect(view).toMatchSnapshot();
    });
});
