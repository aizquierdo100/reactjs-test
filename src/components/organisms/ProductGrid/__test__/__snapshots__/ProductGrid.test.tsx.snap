// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGrid renders ProductGridError correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container justify--around container--wrap--m"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container justify--around container--wrap--m"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProductGrid renders ProductGridLoading correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container justify--around container--wrap--m"
      >
        <div
          class="g--3 g-m--4 g-s--12"
        >
          <img
            alt="spinner"
            src="spinner.gif"
            style="height: 60px; width: 60px;"
          />
        </div>
        <div
          class="g--3 g-m--4 g-s--12"
        >
          <img
            alt="spinner"
            src="spinner.gif"
            style="height: 60px; width: 60px;"
          />
        </div>
        <div
          class="g--3 g-m--4 g-s--12"
        >
          <img
            alt="spinner"
            src="spinner.gif"
            style="height: 60px; width: 60px;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container justify--around container--wrap--m"
    >
      <div
        class="g--3 g-m--4 g-s--12"
      >
        <img
          alt="spinner"
          src="spinner.gif"
          style="height: 60px; width: 60px;"
        />
      </div>
      <div
        class="g--3 g-m--4 g-s--12"
      >
        <img
          alt="spinner"
          src="spinner.gif"
          style="height: 60px; width: 60px;"
        />
      </div>
      <div
        class="g--3 g-m--4 g-s--12"
      >
        <img
          alt="spinner"
          src="spinner.gif"
          style="height: 60px; width: 60px;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ProductGrid renders ProductGridSuccess correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container justify--around container--wrap--m"
      >
        <div
          class="g--3 g-m--4 g-s--12 fade-in-from-top"
        >
          <div
            class="card"
          >
            <img
              alt="Stripe dress"
              class="img-card"
              src="model5.jpg"
            />
            <div
              class="card-content"
            >
              <p
                class="card-title"
              >
                Stripe dress
              </p>
              <p
                class="card-subtitle"
              >
                300,000 VND
              </p>
              <div
                class="container justify--between g--12"
              >
                <button
                  class="btn--raised "
                >
                  More info
                </button>
                <button
                  class="btn--raised "
                >
                  Add to cart
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container justify--around container--wrap--m"
    >
      <div
        class="g--3 g-m--4 g-s--12 fade-in-from-top"
      >
        <div
          class="card"
        >
          <img
            alt="Stripe dress"
            class="img-card"
            src="model5.jpg"
          />
          <div
            class="card-content"
          >
            <p
              class="card-title"
            >
              Stripe dress
            </p>
            <p
              class="card-subtitle"
            >
              300,000 VND
            </p>
            <div
              class="container justify--between g--12"
            >
              <button
                class="btn--raised "
              >
                More info
              </button>
              <button
                class="btn--raised "
              >
                Add to cart
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
